yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -590,6 +605,18 @@ private[spark] class ApplicationMaster(
           case None => logWarning("Container allocator is not ready to kill executors yet.")
         }
         context.reply(true)
+
+      case GetExecutorLossReason(eid) =>
+        Option(allocator) match {
+          case Some(a) =>
+            pendingLossReasonRequests.synchronized {
+              pendingLossReasonRequests
```

Comment:
We can probably check if the executor exists in the executor id map before trying to add to pendingLossReasonRequests to remove the race. Need to be careful about synchronization however.